api/src/main/java/marquez/db/JobDao.java

```diff
@@ -88,6 +88,16 @@ SELECT run_uuid, JSON_AGG(e.facets) AS facets
   """)
   void delete(String namespaceName, String name);
 
+  @SqlUpdate(
+      """
+  UPDATE jobs
+  SET is_hidden = true
+  FROM namespaces n
+  WHERE jobs.namespace_uuid = n.uuid
+  AND n.name = :namespaceName
+  """)
+  void deleteByNamespaceName(String namespaceName);
+
```

Comment:
Shouldn't we also consider job symlinks?

Reply:
Discussed offline. No - because the query captures other symlinks directly, and those are symlinks, not hard links - we don't want to hide them by design.